src/ring_theory/principal_ideal_domain.lean

```diff
@@ -316,6 +316,14 @@ begin
     exacts [gcd_dvd_left x y, gcd_dvd_right x y] },
 end
 
+theorem gcd_dvd_iff_exists (a b : R) {z} : gcd a b ∣ z ↔ ∃ x y, z = a * x + b * y :=
+by simp_rw [mul_comm a, mul_comm b, @eq_comm _ z, ←mem_span_pair, ←span_gcd,
+  ideal.mem_span_singleton]
+
+/-- **Bézout's lemma** -/
+theorem exists_gcd_eq_mul_add_mul (a b : R) : ∃ x y, gcd a b = a * x + b * y :=
+by rw [←gcd_dvd_iff_exists]
```

Comment:
Any reason for adding the `mul_comm`s to give `z = a * x + b * y` instead of `x * a + y * b = z`? Without any external motivation, I'd go for the one with a shorter proof and matches `mem_span_pair`.

Reply:
(alternatively, we could change the phrasing of that lemma instead)

Reply:
@alexjbest [asked](https://github.com/leanprover-community/mathlib/pull/10810#issuecomment-994793551) to match `euclidean_domain.gcd_eq_gcd_ab` and `int.gcd_eq_gcd_ab` above. I don't think we can change `mem_span_pair`, since it doesn't assume commutativity. Changing the `gcd_eq_gcd_ab`s could work, though.